EIPS/eip-3475.md

```diff
@@ -1,428 +1,605 @@
 
-The `"nonce"` is a nonce of the given bond class. This param is for distinctions of the issuing conditions of the bond.
 
-The `"_amount"` is the amount of the bond, that will be transferred from `"_from"`address to `"_to"` address.
+**TransferFrom**
 
-e.g.
+`transferFrom(address _from, address _to, Transaction[] calldata _transaction)` allows the transfer of any number of bond types from an address to another.
 
-```
-transferFrom(0x2d03B6C79B75eE7aB35298878D05fe36DC1fE8Ef,0x82a55a613429Aeb3D01fbE6841bE1AcA4fFD5b2B, 1, 42, 500000000);
-```
+The `"_from"` argument is the address of the holder whose balance about to decrease.
 
-This example shows the transfer from `_from` address, to `\_to` address, `500000000` of bond class`1` nonce `42`
+The `"_to"` argument is the address of the recipient whose balance is about to increase.
 
+The `"_transaction"` is the "`Transaction[] calldata"` structure defined in the rationale.
 
-**Issue**
+@dev transferFrom MUST have the `isApprovedFor(_from, _to, _transaction[i].classId)` approval to transfer `_from` address to `_to` address for given classId (for all transaction in the array structure).
 
-`issue(address to, uint256 classId, uint256 nonceId, uint256 amount)` allows issuing any number of bond types to an address.
+e.g.
 
-The calling of this function needs to be restricted to bond issuer contract.
+``` solidity
+function transferFrom(0x2d03B6C79B75eE7aB35298878D05fe36DC1fE8Ef, 0x82a55a613429Aeb3D01fbE6841bE1AcA4fFD5b2B, [[1, 42, 500000000]]);
+```
 
-The `"to"` argument is the address to which the bond will be issued.
+transfer from `_from` address, to `_to` address, `500000000` bonds of type class`1` and nonce `42`.
 
-The `"classId"` is the classId of the bond, the first bond class created will be 0, and so on.
 
-The `"nonceId"` is the nonceId of the given bond class. This param is for distinctions of the issuing conditions of the bond.
 
-The `"_amount"` is the amount of the bond, that "to" address will receive.
+**transferAllowanceFrom** 
 
-e.g.
+`function transferAllowanceFrom(address _from,address _to, Transaction[] calldata _transactions) public virtual override`
 
-```
-issueBond(0x2d03B6C79B75eE7aB35298878D05fe36DC1fE8Ef, 0, 5, 1000);
-```
+allows the transfer of only those bonds types and nonces which are being alloted to the _to address using allowance() function.
 
-This example shows the issuing to wallet address 0x2d03B6C79B75eE7aB35298878D05fe36DC1fE8Ef, 1000 of bond classId 0 and nonceId 5.
+The `"_from"` argument is the address of the holder whose balance about to decrease.
 
+The `"_to"` argument is the address of the recipient whose balance is about to increase.
 
+The `"_transaction"` is the "`Transaction[] calldata"` structure defined in the section `rationales` below.
 
-**Redeem**
+@dev transferAllowanceFrom MUST have the `allowance(_from, msg.sender, _transactions[i].classId, _transactions[i].nonceId)` implemented inside (where `i` looping for all the transaction index) 
```

Comment:
I'm not sure what the goal of this line is with the `@dev`, but I think it doesn't make sense in plaintext here.